src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -492,7 +492,7 @@ private TypeWithAnnotations ComputeType(Binder binder, SyntaxNode syntax, Bindin
                 allowRefOrOut: false,
                 allowThis: false,
                 addRefReadOnlyModifier: addRefReadOnlyModifier,
-                diagnostics: diagnostics);
+                diagnostics: diagnostics).Cast<SourceParameterSymbol, ParameterSymbol>();
```

Comment:
>.Cast<SourceParameterSymbol, ParameterSymbol>() [](http://example.com/codeflow?start=41&length=47)

We're casting to `ParameterSymbol` in many callers of `MakeParameters()`. Where are we using the change to `SourceParameterSymbol`, and would it make sense to cast that case to `SourceParameterSymbol` instead?

Reply:
> We're casting to `ParameterSymbol` in many callers of `MakeParameters()`. Where are we using the change to `SourceParameterSymbol`, and would it make sense to cast that case to `SourceParameterSymbol` instead?

Is there a specific concern? This is a safe in-place cast. I intentionally changed return type to a more specific type in order to ensure that assumptions made in ```ParameterHelpers.ReportParameterErrors``` are justified.